Tasks/HelmDeployV0/src/helmcommands/helmregistrylogin.ts

```diff
@@ -2,17 +2,35 @@
 
 import tl = require('azure-pipelines-task-lib/task');
 import helmcli from "../helmcli";
+import ACRAuthenticationTokenProvider from "azure-pipelines-tasks-docker-common/registryauthenticationprovider/acrauthenticationtokenprovider";
 
 /*
 Signs into ACR helm registry using service principal
 */
 
-export function addArguments(helmCli: helmcli): void {
-    helmCli.addArgument("login");
+export async function addArguments(helmCli: helmcli): Promise<void> {
     const acrEndpoint = tl.getInput("azureSubscriptionEndpointForACR");
-    const user = tl.getEndpointAuthorizationParameter(acrEndpoint, 'serviceprincipalid', true);
-    const password = tl.getEndpointAuthorizationParameter(acrEndpoint, 'serviceprincipalkey', true);
     const acr = tl.getInput("azureContainerRegistry");
+    const authScheme = tl.getEndpointAuthorizationScheme(acrEndpoint, false).toLowerCase()
+    
+    let user: string, password: string;
+
+    if (authScheme === "workloadidentityfederation") {
+        const tokenProvider = new ACRAuthenticationTokenProvider(acrEndpoint, tl.getInput("azureContainerRegistry"));
+        const token = await tokenProvider.getToken();
+
+        user = token.getUsername();
```

Comment:
According to https://github.com/microsoft/azure-pipelines-tasks-common-packages/blob/main/common-npm-packages/docker-common/registryauthenticationprovider/acrauthenticationtokenprovider.ts#L79C13-L84C46
username will always be the same here since it's hardcoded, only token will be returned as password.
But are you sure that we should use (from lines 33-38 below) for WIF-based authentication in ACR like this?
login %ACR_NAME% --username %user% --password %password%
Please double-check with docs